core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/AbstractPercentilesTestCase.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.metrics.percentiles;
+
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+import org.elasticsearch.search.aggregations.InternalAggregationTestCase;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AbstractPercentilesTestCase<T extends InternalAggregation & Iterable<Percentile>>
+        extends InternalAggregationTestCase<T> {
+
+    private double[] percents;
+    private boolean keyed;
+    private DocValueFormat docValueFormat;
+
+    @Before
+    public void init() {
+        percents = randomPercents();
+        keyed = randomBoolean();
+        docValueFormat = randomNumericDocValueFormat();
+    }
+
+    @Override
+    protected T createTestInstance(String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
+        int numValues = randomInt(100);
+        double[] values = new double[numValues];
+        for (int i = 0; i < numValues; ++i) {
+            values[i] = randomDouble();
+        }
+        return createTestInstance(name, pipelineAggregators, metaData, keyed, docValueFormat, percents, values);
```

Comment:
I'd prefer percents/keyed/docValueFormat to be declared and instantiated here if there is no other reason I am missing atm.